src/DataAPI.jl

```diff
@@ -522,4 +522,14 @@ Return the row number of `row` in the source table.
 """
 function rownumber end
 
+"""
+    groupby(obj, args...; kw...)
+
+Group `obj` into partitions specified by `args` and `kw`,
```

Comment:
As commented in the original issue - I think we should add a comment that `obj` argument must ensure an appropriate type restriction so that there is no type piracy in the implementation of `groupby`.

Reply:
Done